### ISSUE TITLE
Persistent attitude

### ISSUE THREAD

**Issue by gotmachine** [Kerbalism/Kerbalism4#21]
Handling of the vessel orientation for solar power and radiation protection is cumbersome. We could implement a loaded/unloaded/packed agnostic attitude controler UI with the classic 6 orientations, reference body/target selection and even roll control. Could work like that : 

- When loaded, bidirectional interaction with the stock SAS and possibly other attitude controllers (MechJeb, kOS, TCA...)
- When packed, automagically maintain the selected attitude

Could also implement a persistent rotation feature, this is needed so the new artificial gravity feature isn't borked every time you timewarp.

**Comment by alex-dev** [comment]
Principia handles passive rotation accross timewrap and unloading. Ideally any solution would not break passive rotation handling. A spin-stabilized craft relying on persistent solar orientation for power is a pretty common vessel for minimalist probes.

Would the attitude keeping use reaction wheels and RCS or would it rely on the kraken to work?
Would there be a best solar orientation or minimal radiation exposure mode?

**Comment by gotmachine** [comment]
Yep, we will likely disable the persistent rotation feature if Principia is installed.

> Would the attitude keeping use reaction wheels and RCS or would it rely on the kraken to work?

No, as I said, this will be magical. Two reasons : 
- Real life "long term" attitude control schemes are often based on interactions that aren't modeled at all in KSP (gravity gradients, magnetorquers...), and are out of scope for Kerbalism. And applying fixed rate resources consumption "for the sake of it" would be just wrong and frustrating.
- Correctly tracking attitude in the background is a technical nightmare due to the absence of fixed reference frame in KSP.

> Would there be a best solar orientation or minimal radiation exposure mode?

No. up to the player to use the one he see fit. We are providing the relevant information. If we provide the 6 base directions + target/antiTarget, the ability to freely define the reference body and a separate reference for roll control, I think this should be more than enough for handling any possible vessel design.

**Comment by Fortran-to-Python** [comment]
Persist Thrust and Persistent Rotation already take care of this.

**Comment by alex-dev** [comment]
Persist Thrust seems to be dead since 1.10. And it is unrelated to attitude control. More geared toward unloaded propulsion. Which is very likely to not work with Kerbalism overhauling resource management.
Persistent Rotation could work. Depends on how it is implemented and if Kerbalism can leverage it for rotation dependant stuff (artificial gravity) when unloaded.

**Comment by Fortran-to-Python** [comment]
It may not have gotten updated since 1.10, but it still works (mostly).
